deepinv/training/trainer.py

```diff
@@ -95,7 +95,8 @@ class Trainer:
         Should be used in conjunction with ``online_measurements=True``, no effect when ``online_measurements=False``. Also see ``loop_random_online_physics``.
     :param bool loop_random_online_physics: if True, resets the physics generator **and** noise model back to its initial state at the beginning of each epoch,
         so that the same measurements are generated each epoch. Requires `shuffle=False` in dataloaders. If False, generates new physics every epoch.
-        Used in conjunction with ``physics_generator`` and ``online_measurements=True``, no effect when ``online_measurements=False``.
+        Used in conjunction with ``physics_generator``.
+    :param bool global_optimizer_step: If ``True``, the optimizer step is performed once on all datasets. If ``False``, the optimizer step is performed on each dataset separately.
```

Comment:
Please can we add a test for this? (well, mostly for when `global_optimizer_step=False` that the original behaviour is recovered, and that global_optimizer_step does nothing when there is only one dataset